streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -0,0 +1,202 @@
+
+  /**
+   * If `Serialized[K, V]` is found in the implicit scope, then use it, else
+   * use the API with the default serializers.
+   *
+   * Usage Pattern 1: No implicits in scope, use default serializers
+   * - .groupByKey
+   *
+   * Usage Pattern 2: Use implicit `Serialized` in scope
+   * implicit val serialized = Serialized.`with`(stringSerde, longSerde)
+   * - .groupByKey
+   *
+   * Usage Pattern 3: uses the implicit conversion from the serdes to `Serialized`
+   * implicit val stringSerde: Serde[String] = Serdes.String()
+   * implicit val longSerde: Serde[Long] = Serdes.Long().asInstanceOf[Serde[Long]]
+   * - .groupByKey
+   */
+  def groupByKey(implicit serialized: Serialized[K, V]): KGroupedStream[K, V] =
+    inner.groupByKey(serialized)
+
+  def groupBy[KR](selector: (K, V) => KR)(implicit serialized: Serialized[KR, V]): KGroupedStream[KR, V] =
+    inner.groupBy(selector.asKeyValueMapper, serialized)
+
+  def join[VO, VR](otherStream: KStream[K, VO],
+    joiner: (V, VO) => VR,
+    windows: JoinWindows)(implicit joined: Joined[K, V, VO]): KStream[K, VR] =
+      inner.join[VO, VR](otherStream.inner, joiner.asValueJoiner, windows, joined)
+
+  def join[VT, VR](table: KTable[K, VT],
+    joiner: (V, VT) => VR)(implicit joined: Joined[K, V, VT]): KStream[K, VR] =
+      inner.leftJoin[VT, VR](table.inner, joiner.asValueJoiner, joined)
+
+  def join[GK, GV, RV](globalKTable: GlobalKTable[GK, GV],
+    keyValueMapper: (K, V) => GK,
+    joiner: (V, GV) => RV): KStream[K, RV] =
+      inner.join[GK, GV, RV](globalKTable, keyValueMapper(_,_), joiner(_,_))
+
+  def leftJoin[VO, VR](otherStream: KStream[K, VO],
+    joiner: (V, VO) => VR,
+    windows: JoinWindows)(implicit joined: Joined[K, V, VO]): KStream[K, VR] =
+      inner.leftJoin[VO, VR](otherStream.inner, joiner.asValueJoiner, windows, joined)
+
+  def leftJoin[VT, VR](table: KTable[K, VT],
+    joiner: (V, VT) => VR)(implicit joined: Joined[K, V, VT]): KStream[K, VR] =
+      inner.leftJoin[VT, VR](table.inner, joiner.asValueJoiner, joined)
+
+  def leftJoin[GK, GV, RV](globalKTable: GlobalKTable[GK, GV],
+    keyValueMapper: (K, V) => GK,
+    joiner: (V, GV) => RV): KStream[K, RV] = {
+
+    inner.leftJoin[GK, GV, RV](globalKTable, keyValueMapper.asKeyValueMapper, joiner.asValueJoiner)
+  }
+
+  def outerJoin[VO, VR](otherStream: KStream[K, VO],
+    joiner: (V, VO) => VR,
+    windows: JoinWindows)(implicit joined: Joined[K, V, VO]): KStream[K, VR] =
+      inner.outerJoin[VO, VR](otherStream.inner, joiner.asValueJoiner, windows, joined)
+
+  def merge(stream: KStream[K, V]): KStream[K, V] = inner.merge(stream)
+
+  def peek(action: (K, V) => Unit): KStream[K, V] = {
+    inner.peek(action(_,_))
+  }
+
+  // -- EXTENSIONS TO KAFKA STREAMS --
+
+  // applies the predicate to know what messages should go to the left stream (predicate == true)
+  // or to the right stream (predicate == false)
+  def split(predicate: (K, V) => Boolean): (KStream[K, V], KStream[K, V]) = {
```

Comment:
I agree that both APIs should be consistent. Does a `split` add much value compare to `branch`? Btw, this might be related to https://issues.apache.org/jira/browse/KAFKA-5488

I am also open to add a `split()` if we think it's useful.

Reply:
I agree.

Reply:
For the time being, we can remove `split` from the Scala API and rethink if / when it's implemented as a Java API

Reply:
Sounds good.

Reply:
ok, will remove `split` from `KStream` for now.

Reply:
Removed!